src/tracks/labeltrack/ui/LabelTrackView.cpp

```diff
@@ -877,46 +829,20 @@ void LabelTrackView::Draw
       }
    }
 
-   // Draw highlights
-   if ( (mInitialCursorPos != mCurrentCursorPos) && IsValidIndex(mTextEditIndex, project))
-   {
-      int xpos1, xpos2;
-      CalcHighlightXs(&xpos1, &xpos2);
-      DrawHighlight(dc, mLabels[mTextEditIndex],
-         xpos1, xpos2, dc.GetFontMetrics().ascent + dc.GetFontMetrics().descent);
-   }
-
    // Draw the text and the label boxes.
-   { int i = -1; for (const auto &labelStruct : mLabels) { ++i;
-      if(mTextEditIndex == i )
-         dc.SetBrush(AColor::labelTextEditBrush);
-      DrawText( dc, labelStruct, r );
-      if(mTextEditIndex == i )
-         dc.SetBrush(AColor::labelTextNormalBrush);
-   }}
-
-   // Draw the cursor, if there is one.
-   if(mInitialCursorPos == mCurrentCursorPos && IsValidIndex(mTextEditIndex, project))
+   int i = 0;
+   for (auto label : mLabels)
    {
-      const auto &labelStruct = mLabels[mTextEditIndex];
-      int xPos = labelStruct.xText;
-
-      if( mCurrentCursorPos > 0)
+      if ((i == mTextEditIndex) && mTextEditHelper)
       {
-         // Calculate the width of the substring and add it to Xpos
-         int partWidth;
-         dc.GetTextExtent(labelStruct.title.Left(mCurrentCursorPos), &partWidth, NULL);
-         xPos += partWidth;
+         mTextEditHelper->Draw(dc, LabelTrackView::MakeTextEditHelperRect(label));
       }
-
-      wxPen currentPen = dc.GetPen();
-      const int CursorWidth=2;
-      currentPen.SetWidth(CursorWidth);
-      const auto top = labelStruct.y - (LabelBarHeight + yFrameHeight) / 2 + (yFrameHeight - mFontHeight) / 2 + TextFrameYOffset;
-      AColor::Line(dc,
-                   xPos-1, top,
-                   xPos-1, top + mFontHeight);
-      currentPen.SetWidth(1);
+      else
+      {
+         dc.SetBrush(AColor::labelTextNormalBrush);
```

Comment:
Are you sure `labelTextEditBrush` is used for the highlighted text box?  It's not obvious to me whether these state changes in `dc` are improperly affecting what `mTextEditHelper->Draw` does.